### PR TITLE
fix white symbol in price/ fix special symbol in attribute value

### PR DESCRIPTION
**Pull request by MrPogos**

### CODE REVIEW

File: src/EventListener/ResourceIndexListener.php [hunk 1]
```diff
@@ -44,6 +53,14 @@ public function updateIndex(GenericEvent $event): void
             if ($resource instanceof $config[self::MODEL_KEY]) {
                 $this->resourceRefresher->refresh($resource, $config[self::SERVICE_ID_KEY]);
             }
+
+            if ($resource instanceof Product
+                && 'Sylius\Component\Product\Model\ProductAttribute' === $config[self::MODEL_KEY]
+            ) {
```
Comment (senghe) [inline src/EventListener/ResourceIndexListener.php:59]:
```suggestion
            if ($resource instanceof Product
                && Sylius\Component\Product\Model\ProductAttribute::class === $config[self::MODEL_KEY]
            ) {
```

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateMapper.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        return self::ATTRIBUTES_TYPE === $type;
+    }
+
+    public function map(array $attributeValues): array
+    {
+        $choices = [];
+
+        foreach ($attributeValues as $productAttributeValue) {
+            $choice = $productAttributeValue['value']->format('d-m-Y H:i');
+            $value = $productAttributeValue['value']->format('Y-m-d H:i:s.sss');
```
Comment (senghe) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateMapper.php:29]: Is the `.sss` needed?

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return self::ATTRIBUTES_TYPE === $type;
+    }
+
+    public function map(array $attributeValues): array
+    {
+        $choices = [];
+
+        foreach ($attributeValues as $productAttributeValue) {
+            $choice = $productAttributeValue['value']->format('d-m-Y H:i');
+            $value = $productAttributeValue['value']->format('Y-m-d H:i:s.sss');
```
Comment (senghe) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php:29]: Is `.sss` needed?

File: src/PropertyBuilder/AttributeBuilder.php [hunk 4]
```diff
@@ -15,13 +15,13 @@
...
 use Sylius\Component\Attribute\Model\AttributeInterface;
-use Sylius\Component\Attribute\Model\AttributeTranslation;
 use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Product\Model\ProductAttributeValue;
 
 final class AttributeBuilder extends AbstractBuilder
 {
     /** @var ConcatedNameResolverInterface */
-    private $attributeNameResolver;
+    private $attributeNzzzameResolver;
```
Comment (senghe) [inline src/PropertyBuilder/AttributeBuilder.php:24]: This name seems to be broken ;)

File: spec/PropertyBuilder/AttributeBuilderSpec.php [hunk 5]
```diff
@@ -22,7 +22,7 @@
 
 final class AttributeBuilderSpec extends ObjectBehavior
```
Comment (senghe) [inline spec/PropertyBuilder/AttributeBuilderSpec.php:23]: The production class has been changed, but there is no change in spec file. Could you check all classes you changed by this?

File: src/PropertyBuilder/AttributeBuilder.php [hunk 6]
```diff
@@ -89,21 +88,22 @@ private function resolveProductAttribute(
...
-            $code = \sprintf('%s_%s', $documentKey, $attributeTranslation->getLocale());
-            $values = $this->resolveProductAttribute(
-                $attributeConfiguration,
-                $value,
-                $attributeTranslation
-            );
-            $document->set($code, $values);
-        }
+
+        $value = $productAttribute->getValue();
```
Comment (senghe) [inline src/PropertyBuilder/AttributeBuilder.php:97]: There is no spec for this

File: src/PropertyBuilder/AttributeTaxonsBuilder.php [hunk 7]
```diff
@@ -40,6 +41,7 @@ public function consumeEvent(PostTransformEvent $event): void
         $documentAttribute = $event->getObject();
 
         if (!$documentAttribute instanceof AttributeInterface
+            || !$documentAttribute instanceof ProductAttributeInterface
```
Comment (senghe) [inline src/PropertyBuilder/AttributeTaxonsBuilder.php:44]: There is no spec for this change

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeDateQueryBuilder.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    public function supports(string $type): bool
+    {
+        return in_array($type, self::AVAILABLE_ATTRIBUTES_TYPE);
+    }
+
+    public function buildQuery(array $data, string $localCode): BoolQuery
+    {
+        $attributeQuery = new BoolQuery();
+
+        foreach ((array) $data['attribute_values'] as $attributeValue) {
```
Comment (senghe) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeDateQueryBuilder.php:33]: I would put assertions for all keys you use in the array

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    public function supports(string $type): bool
+    {
+        return in_array($type, self::AVAILABLE_ATTRIBUTES_TYPE);
+    }
+
+    public function buildQuery(array $data, string $localCode): BoolQuery
+    {
+        $attributeQuery = new BoolQuery();
+
+        foreach ((array) $data['attribute_values'] as $attributeValue) {
```
Comment (senghe) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php:34]: Assertions for arrays pls

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\QueryBuilder\AttributesQueryBuilder;
+
+use Elastica\Query\BoolQuery;
+use Elastica\Query\Term;
+
+class AttributesTypeNumberQueryBuilder implements AttributesQueryBuilderCollectorInterface
```
Comment (senghe) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php:17]: I would rename all classes inside using a rule as below:

```suggestion
class NumberAttributesQueryBuilder implements AttributesQueryBuilderCollectorInterface
```

Comment (senghe) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php:17]: And could you take care on finalizing all classes?

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeTextQueryBuilder.php [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    public function supports(string $type): bool
+    {
+        return in_array($type, self::AVAILABLE_ATTRIBUTES_TYPE);
+    }
+
+    public function buildQuery(array $data, string $localCode): BoolQuery
+    {
+        $attributeQuery = new BoolQuery();
+
+        foreach ((array) $data['attribute_values'] as $attributeValue) {
```
Comment (senghe) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeTextQueryBuilder.php:34]: Assertions pls

File: src/Repository/ProductAttributeRepository.php [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+{
+    /** @var RepositoryInterface */
+    private $productAttributeRepository;
+
+    public function __construct(RepositoryInterface $productAttributeRepository)
+    {
+        $this->productAttributeRepository = $productAttributeRepository;
+    }
+
+    public function getAttributeTypeByName(string $attributeName): string
```
Comment (senghe) [inline src/Repository/ProductAttributeRepository.php:26]: Could you write integration tests for this method?

File: src/Repository/ProductAttributeValueRepository.php [hunk 13]
```diff
@@ -23,7 +24,7 @@ public function __construct(BaseAttributeValueRepositoryInterface $baseAttribute
         $this->baseAttributeValueRepository = $baseAttributeValueRepository;
     }
 
-    public function getUniqueAttributeValues(AttributeInterface $productAttribute): array
+    public function getUniqueAttributeValues(AttributeInterface $productAttribute, Taxon $taxon): array
```
Comment (senghe) [inline src/Repository/ProductAttributeValueRepository.php:27]: Could you write integration tests for this method?

File: src/Repository/ProductAttributeValueRepository.php [hunk 14]
```diff
@@ -12,6 +12,7 @@
 
 use Sylius\Component\Attribute\Model\AttributeInterface;
 use Sylius\Component\Product\Repository\ProductAttributeValueRepositoryInterface as BaseAttributeValueRepositoryInterface;
+use Sylius\Component\Taxonomy\Model\Taxon;
 
 class ProductAttributeValueRepository implements ProductAttributeValueRepositoryInterface
```
Comment (senghe) [inline src/Repository/ProductAttributeValueRepository.php:17]: Would you make this class final?

File: src/Resources/config/services/form.xml [hunk 15]
```diff
@@ -53,5 +55,17 @@
             <argument type="service" id="bitbag_sylius_elasticsearch_plugin.facet.registry" />
             <tag name="form.type" />
         </service>
+
+        <service id="bitbag_sylius_elasticsearch_plugin.form.mapper.type.data" class="BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper\AttributesTypeDateMapper">
```
Comment (senghe) [inline src/Resources/config/services/form.xml:59]:
```suggestion
        <service id="bitbag_sylius_elasticsearch_plugin.form.mapper.type.date" class="BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper\AttributesTypeDateMapper">
```

File: src/Resources/config/services/form.xml [hunk 16]
```diff
@@ -53,5 +55,17 @@
...
+            <tag name="bitbag_sylius_elasticsearch_plugin.form.type.product_attributes_mapper" />
+        </service>
+
+        <service id="bitbag_sylius_elasticsearch_plugin.form.mapper.type.datatype" class="BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper\AttributesTypeDateTimeMapper">
```
Comment (senghe) [inline src/Resources/config/services/form.xml:63]:
```suggestion
        <service id="bitbag_sylius_elasticsearch_plugin.form.mapper.type.datetime" class="BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper\AttributesTypeDateTimeMapper">
```

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateMapper.php (hunk 2 above)
Comment (MrPogos) [reply src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateMapper.php:29]: In order not to file it's full date format, elasticsearch stores our date in this version 2017-01-01 00: 13: 00.000000

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateMapper.php [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        return self::ATTRIBUTES_TYPE === $type;
+    }
+
+    public function map(array $attributeValues): array
+    {
+        $choices = [];
+
+        foreach ($attributeValues as $productAttributeValue) {
+            $choice = $productAttributeValue['value']->format('d-m-Y H:i');
```
Comment (christopherhero) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateMapper.php:28]: move formats to const

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesMapperCollectorInterface.php [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+ * Feel free to contact us once you face any issues or want to start
+ * You can find more information about us on https://bitbag.io and write us
+ * an email on hello@bitbag.io.
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper;
+
+interface AttributesMapperCollectorInterface
```
Comment (christopherhero) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesMapperCollectorInterface.php:14]: why don't you add unit tests for mappers?

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+ * Feel free to contact us once you face any issues or want to start
+ * You can find more information about us on https://bitbag.io and write us
+ * an email on hello@bitbag.io.
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper;
+
+class AttributesTypeDateTimeMapper implements AttributesMapperCollectorInterface
```
Comment (christopherhero) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php:14]: final

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        return self::ATTRIBUTES_TYPE === $type;
+    }
+
+    public function map(array $attributeValues): array
+    {
+        $choices = [];
+
+        foreach ($attributeValues as $productAttributeValue) {
+            $choice = $productAttributeValue['value']->format('d-m-Y H:i');
```
Comment (christopherhero) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php:28]: please move formats to consts

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypePercentMapper.php [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+ * Feel free to contact us once you face any issues or want to start
+ * You can find more information about us on https://bitbag.io and write us
+ * an email on hello@bitbag.io.
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\Form\Type\ChoiceMapper\AttributesMapper;
+
+class AttributesTypePercentMapper implements AttributesMapperCollectorInterface
```
Comment (christopherhero) [inline src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypePercentMapper.php:14]: final
Composition over inheritance ;d

File: src/Form/Type/ChoiceMapper/ProductAttributesMapper.php [hunk 22]
```diff
@@ -52,8 +64,14 @@ public function mapToChoices(ProductAttributeInterface $productAttribute): array
...
         }
+        $taxon = $this->taxonContext->getTaxon();
+        $attributeValues = $this->productAttributeValueRepository->getUniqueAttributeValues($productAttribute, $taxon);
 
-        $attributeValues = $this->productAttributeValueRepository->getUniqueAttributeValues($productAttribute);
+        foreach ($this->attributeMapper as $mapper) {
+            if ($mapper->supports($productAttribute->getType())) {
+                return $mapper->map($attributeValues);
+            }
+        }
```
Comment (christopherhero) [inline src/Form/Type/ChoiceMapper/ProductAttributesMapper.php:74]: what with the rest of code below ?

File: src/Form/Type/PriceFilterType.php [hunk 23]
```diff
@@ -62,6 +64,15 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     ]),
                 ],
             ])
+            ->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
+                if (!empty($event->getData())) {
+                    $data = [];
+                    foreach ($event->getData() as $key => $item) {
```
Comment (christopherhero) [inline src/Form/Type/PriceFilterType.php:70]: add some spaces

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeDateQueryBuilder.php [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\QueryBuilder\AttributesQueryBuilder;
+
+use Elastica\Query\BoolQuery;
+use Elastica\Query\Term;
+
+class AttributesTypeDateQueryBuilder implements AttributesQueryBuilderCollectorInterface
```
Comment (christopherhero) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeDateQueryBuilder.php:17]: final

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeTextQueryBuilder.php [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\QueryBuilder\AttributesQueryBuilder;
+
+use Elastica\Query\BoolQuery;
+use Elastica\Query\Term;
+
+class AttributesTypeTextQueryBuilder implements AttributesQueryBuilderCollectorInterface
```
Comment (christopherhero) [inline src/QueryBuilder/AttributesQueryBuilder/AttributesTypeTextQueryBuilder.php:17]: final

File: src/QueryBuilder/HasAttributesQueryBuilder.php [hunk 26]
```diff
@@ -10,31 +10,43 @@
...
+    ) {
         $this->localeContext = $localeContext;
+        $this->productAttributeRepository = $productAttributeRepository;
+        $this->attributeDriver = $attributeDriver;
     }
 
     public function buildQuery(array $data): ?AbstractQuery
     {
-        $attributeQuery = new BoolQuery();
+        $attributeName = str_replace('attribute_', '', $data['attribute']);
```
Comment (christopherhero) [inline src/QueryBuilder/HasAttributesQueryBuilder.php:40]: Are you sure you need it ? I think not 
if you need an alias for attribute name you can use simply `$attributeName = $data['attribute']`

File: src/Resources/config/services/query_builder.xml [hunk 27]
```diff
@@ -33,6 +33,20 @@
 
         <service id="bitbag_sylius_elasticsearch_plugin.query_builder.has_attributes" class="BitBag\SyliusElasticsearchPlugin\QueryBuilder\HasAttributesQueryBuilder" >
             <argument type="service" id="sylius.context.locale"/>
+            <argument type="service" id="bitbag.sylius_elasticsearch_plugin.repository.product_attribute_repository"/>
+            <argument type="tagged_iterator" tag="bitbag_sylius_elasticsearch_plugin.query_builder.attributes" />
+        </service>
+
+        <service id="bitbag_sylius_elasticsearch_plugin.query_builder.attributes.type.text" class="BitBag\SyliusElasticsearchPlugin\QueryBuilder\AttributesQueryBuilder\AttributesTypeTextQueryBuilder">
```
Comment (christopherhero) [inline src/Resources/config/services/query_builder.xml:40]: try to use formatting that is more pleasing to the eyes, e.g:
```
    <service id="bitbag_sylius_elasticsearch_plugin.query_builder.attributes.type.text" 
             class="BitBag\SyliusElasticsearchPlugin\QueryBuilder\AttributesQueryBuilder\AttributesTypeTextQueryBuilder"
    >```

File: tests/PHPUnit/Integration/IntegrationTestCase.php [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+namespace Tests\BitBag\SyliusElasticsearchPlugin\Integration;
+
+
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/IntegrationTestCase.php:12]: remove space here and  add after line 9

File: tests/PHPUnit/Integration/IntegrationTestCase.php [hunk 29]
```diff
@@ -0,0 +1,33 @@
...
+use ApiTestCase\JsonApiTestCase;
+
+abstract class IntegrationTestCase extends JsonApiTestCase
+{
+    public function __construct(?string $name = null, array $data = [], string $dataName = '')
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/IntegrationTestCase.php:17]: split arguments

File: tests/PHPUnit/Integration/IntegrationTestCase.php [hunk 30]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        parent::__construct($name, $data, $dataName);
+
+        $this->dataFixturesPath = __DIR__ . '/DataFixtures/ORM';
+    }
+
+    protected function setUp(): void
+    {
+
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/IntegrationTestCase.php:26]: remove space or add `return;`

File: tests/PHPUnit/Integration/IntegrationTestCase.php [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    protected function tearDown(): void
+    {
+
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/IntegrationTestCase.php:31]: remove space or add `return;`

File: tests/PHPUnit/Integration/Repository/ProductAttributeRepositoryTest.php [hunk 32]
```diff
@@ -0,0 +1,39 @@
...
+namespace Tests\BitBag\SyliusElasticsearchPlugin\Integration\Repository;
+
+use BitBag\SyliusElasticsearchPlugin\Repository\ProductAttributeRepository;
+use Tests\BitBag\SyliusElasticsearchPlugin\Integration\IntegrationTestCase;
+
+class ProductAttributeRepositoryTest extends IntegrationTestCase
+{
+    /** @var ProductAttributeRepository */
+    private $attributeRepository;
+
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/Repository/ProductAttributeRepositoryTest.php:21]: remove space

File: tests/PHPUnit/Integration/Repository/ProductAttributeValueRepositoryTest.php [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+use BitBag\SyliusElasticsearchPlugin\Repository\ProductAttributeValueRepository;
+use BitBag\SyliusElasticsearchPlugin\Repository\TaxonRepository;
+use Sylius\Bundle\CoreBundle\Doctrine\ORM\AttributeRepository;
+use Tests\BitBag\SyliusElasticsearchPlugin\Integration\IntegrationTestCase;
+
+class ProductAttributeValueRepositoryTest extends IntegrationTestCase
+{
+    /** @var ProductAttributeValueRepository */
+    private $productAttributeValueRepository;
+
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/Repository/ProductAttributeValueRepositoryTest.php:16]: remove space (please check all PR)

File: tests/PHPUnit/Integration/Repository/ProductAttributeValueRepositoryTest.php [hunk 34]
```diff
@@ -0,0 +1,44 @@
...
+    public function SetUp(): void
+    {
+        parent::SetUp();
+
+        $this->productAttributeValueRepository = self::$container->get('bitbag.sylius_elasticsearch_plugin.repository.product_attribute_value_repository');
+        $this->attributeRepository = self::$container->get('sylius.repository.product_attribute');
+        $this->taxonRepository = self::$container->get('sylius.repository.taxon');
+    }
+    public function test_get_unique_attribute_values(): void
```
Comment (christopherhero) [inline tests/PHPUnit/Integration/Repository/ProductAttributeValueRepositoryTest.php:32]: add space before

**Review by christopherhero — CHANGES_REQUESTED** [review]

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeDateQueryBuilder.php (hunk 8 above)
Comment (senghe) [reply src/QueryBuilder/AttributesQueryBuilder/AttributesTypeDateQueryBuilder.php:33]: It still pops up

File: src/PropertyBuilder/AttributeTaxonsBuilder.php (hunk 7 above)
Comment (senghe) [reply src/PropertyBuilder/AttributeTaxonsBuilder.php:44]: It still pops up

File: src/PropertyBuilder/AttributeBuilder.php (hunk 6 above)
Comment (senghe) [reply src/PropertyBuilder/AttributeBuilder.php:97]: It still pops up

File: spec/PropertyBuilder/AttributeBuilderSpec.php (hunk 5 above)
Comment (senghe) [reply spec/PropertyBuilder/AttributeBuilderSpec.php]: It still pops up

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php (hunk 9 above)
Comment (senghe) [reply src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php:34]: It still pops up

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php (hunk 10 above)
Comment (senghe) [reply src/QueryBuilder/AttributesQueryBuilder/AttributesTypeNumberQueryBuilder.php:17]: It still pops up

File: src/QueryBuilder/AttributesQueryBuilder/AttributesTypeTextQueryBuilder.php (hunk 11 above)
Comment (senghe) [reply src/QueryBuilder/AttributesQueryBuilder/AttributesTypeTextQueryBuilder.php:34]: It still pops up

File: src/Resources/config/services/form.xml (hunk 16 above)
Comment (senghe) [reply src/Resources/config/services/form.xml]: It still pops up

**Review by senghe — CHANGES_REQUESTED** [review]
Still there is some comments missed by @MrPogos.

File: spec/PropertyBuilder/AttributeBuilderSpec.php (hunk 5 above)
Comment (MrPogos) [reply spec/PropertyBuilder/AttributeBuilderSpec.php]: the changed part of the code is not reflected in the spec tests

File: tests/PHPUnit/Integration/Repository/ProductAttributeValueRepositoryTest.php [hunk 35]
```diff
@@ -0,0 +1,50 @@
+<?php
```
Comment (senghe) [inline tests/PHPUnit/Integration/Repository/ProductAttributeValueRepositoryTest.php:1]:
```suggestion
<?php

/*
 * This file has been created by developers from BitBag.
 * Feel free to contact us once you face any issues or want to start
 * You can find more information about us on https://bitbag.io and write us
 * an email on hello@bitbag.io.
 */
```

**Review by senghe — APPROVED** [review]

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php (hunk 19 above)
Comment (christopherhero) [reply src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php:14]: has not been changed

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php (hunk 20 above)
Comment (christopherhero) [reply src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypeDateTimeMapper.php:28]: please, use this const here

File: src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypePercentMapper.php (hunk 21 above)
Comment (christopherhero) [reply src/Form/Type/ChoiceMapper/AttributesMapper/AttributesTypePercentMapper.php:14]: has not been changed

**Pull request merged**